### PR TITLE
Add support for passing base path for control plane operations

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

Currently users don't have an option to pass in the base path for control plane operations.  

## Solution

Add `host` and `enableTls` fields to Pinecone's inner class called `Builder` and further set the same values for `PineconeConfig`. This allows the same configuration to be used for data plane operations, eliminating the need for users to manually instantiate `PineconeConfig` and pass it to the `Pinecone` class, as was required in the previous approach..

Existing approach for data plane operations:
```java
// Instantiate ManageIndexesApi class
ManageIndexesApi manageIndexesApi = new ManageIndexesApi();

// Instantiate PineconeConfig class
PineconeConfig config = new PineconeConfig("apiKey");
// Disable tls for gRPC calls
config.setTLSEnabled(false);
// Set host for gRPC calls
config.setHost(host);

// Instantiate Pinecone class using PineconeConfig and ManageIndexesApi
Pinecone pinecone = new Pinecone(config, manageIndexesApi);

// Get index connection object for the index
Index index = pinecone.getIndexConnection("index");

// Upsert records into the index
index.upsert("v1", Arrays.asList(1f, 2f, 3f));
```

New approach for control and data plane operations:
```java
// Sets the host for both control and data plane operations
Pinecone pinecone = new Pinecone.Builder("apiKey")
       .withHost(host)
       .withTlsEnabled(false)
       .build();

// Create index (control plane operation)
pinecone.createServerlessIndex("index", "cosine", 3, "aws","us-east-1", DeletionProtection.DISABLED);

// Get index connection object for the index
Index index = pinecone.getIndexConnection("index");

// Upsert records into the index (data plane operation)
index.upsert("v1", Arrays.asList(1f, 2f, 3f));
```
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Existing integration tests should work as it and tested locally using the emulator.

### CODE REVIEW

File: src/main/java/io/pinecone/clients/Pinecone.java [hunk 1]
```diff
@@ -1023,6 +1025,16 @@ public Builder withProxy(String proxyHost, int proxyPort) {
...
+        public Builder withHost(String host) {
+            this.host = host;
+            return this;
+        }
+
+        public Builder withTlsEnabled(boolean enableTls) {
+            this.enableTls = enableTls;
+            return this;
+        }
+
```
Comment (austin-denoble) [inline src/main/java/io/pinecone/clients/Pinecone.java:1037]: Should these have some sort of documentation similar to `withProxy` above?

File: src/main/java/io/pinecone/clients/Pinecone.java [hunk 2]
```diff
@@ -1034,13 +1046,23 @@ public Builder withProxy(String proxyHost, int proxyPort) {
...
 
-            ApiClient apiClient = (customOkHttpClient != null) ? new ApiClient(customOkHttpClient) : new ApiClient(buildOkHttpClient(proxyConfig));
+            ApiClient apiClient;
+            if (customOkHttpClient != null) {
+                apiClient = new ApiClient(customOkHttpClient);
+            } else {
+                apiClient = new ApiClient(buildOkHttpClient(proxyConfig));
+                if(host!=null && !host.isEmpty()) {
+                    config.setHost(host);
+                    apiClient.setBasePath(host);
```
Comment (austin-denoble) [inline src/main/java/io/pinecone/clients/Pinecone.java:1063]: If the host is applied to the `apiClient` does it also need to be passed to `config` here? From what it looks like, this value is overwritten when you target an index: https://github.com/pinecone-io/pinecone-java-client/blob/main/src/main/java/io/pinecone/clients/Pinecone.java#L844-L874

File: src/main/java/io/pinecone/clients/Pinecone.java [hunk 3]
```diff
@@ -1034,13 +1046,23 @@ public Builder withProxy(String proxyHost, int proxyPort) {
          */
         public Pinecone build() {
             PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig, customOkHttpClient);
+            config.setTLSEnabled(enableTls);
```
Comment (austin-denoble) [inline src/main/java/io/pinecone/clients/Pinecone.java:1049]: So this is only needed for `PineconeConfig` and we don't need it for the `ApiClient` itself?

**Review by austin-denoble — APPROVED** [review]
This looks ok to me, but I had a couple questions since it's been a while since I've looked at this codebase.

File: src/main/java/io/pinecone/clients/Pinecone.java (hunk 1 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/clients/Pinecone.java:1037]: Yes I need to the docstrings

File: src/main/java/io/pinecone/clients/Pinecone.java (hunk 3 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/clients/Pinecone.java:1049]: Yes, this enableTls flag is only needed for gRPC connection so the pineconeConfig object is set and further fetched at the time of establishing gRPC connection.

File: src/main/java/io/pinecone/clients/Pinecone.java (hunk 2 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/clients/Pinecone.java:1063]: Currently we are making an extra describeIndex call using the same Pinecone object so the overwritten value is still pointing to the same host but this extra call can be removed. The only approach I came up with for now, in order to solve this, is instantiating the PineconeConfig object outside of Builder class which I don't like it. So I have a ticket to remove this extra call in future and come up with a better solution. Also the ticket will take care of asyncIndexConnection as well.

**Pull request merged**